plugin-server/src/main/ingestion-queues/session-recording/session-recordings-blob-consumer.ts

```diff
@@ -212,33 +216,29 @@ export class SessionRecordingBlobIngester {
             return statusWarn('no_token')
         }
 
-        let team: Team | null = null
+        let teamId: TeamId | null = null
+        const token = messagePayload.token
 
         const teamSpan = span?.startChild({
             op: 'fetchTeam',
         })
-        if (messagePayload.team_id != null) {
-            team = await this.teamManager.fetchTeam(messagePayload.team_id)
-        } else if (messagePayload.token) {
-            team = await this.teamManager.getTeamByToken(messagePayload.token)
+        if (token) {
+            teamId = await this.teamsRefresher.get().then((teams) => teams[token] || null)
         }
         teamSpan?.finish()
 
-        if (team == null) {
-            return statusWarn('team_not_found', {
-                teamId: messagePayload.team_id,
-                payloadTeamSource: messagePayload.team_id ? 'team' : messagePayload.token ? 'token' : 'unknown',
-            })
-        }
-
-        if (!team.session_recording_opt_in) {
+        if (teamId == null) {
             eventDroppedCounter
                 .labels({
                     event_type: 'session_recordings_blob_ingestion',
-                    drop_cause: 'disabled',
+                    drop_cause: 'team_missing_or_disabled',
                 })
                 .inc()
-            return
+
+            return statusWarn('team_missing_or_disabled', {
```

Comment:
it's a shame to lose the separation of signal here but it does reduce storage a good chunk